x-pack/packages/kbn-elastic-assistant/impl/assistant/conversations/conversation_settings_management/use_conversations_table.tsx

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback } from 'react';
+import { ActionTypeRegistryContract } from '@kbn/triggers-actions-ui-plugin/public';
+import { EuiBadge, EuiBasicTableColumn, EuiLink } from '@elastic/eui';
+
+import { FormattedDate } from '@kbn/i18n-react';
+import { Conversation } from '../../../assistant_context/types';
+import { AIConnector } from '../../../connectorland/connector_selector';
+import { getConnectorTypeTitle } from '../../../connectorland/helpers';
+import { Prompt } from '../../../..';
+import { getConversationApiConfig } from '../../use_conversation/helpers';
+import * as i18n from './translations';
+import { RowActions } from '../../common/components/assistant_settings_management/row_actions';
+import { SystemPromptColumn } from './system_prompt_column';
+
+const emptyConversations = {};
+
+export interface GetConversationsListParams {
+  allSystemPrompts: Prompt[];
+  actionTypeRegistry: ActionTypeRegistryContract;
+  connectors: AIConnector[] | undefined;
+  conversations: Record<string, Conversation>;
+  defaultConnector?: AIConnector;
+}
+
+export type ConversationTableItem = Conversation & {
+  actionType?: string | null;
+};
+
+export const useConversationsTable = () => {
+  const getColumns = useCallback(
+    ({
+      onDeleteActionClicked,
+      onEditActionClicked,
+      allSystemPrompts,
+    }): Array<EuiBasicTableColumn<ConversationTableItem>> => {
+      return [
+        {
+          name: i18n.CONVERSATIONS_TABLE_COLUMN_CONVERSATIONS,
+          render: (conversation: ConversationTableItem) => (
+            <EuiLink onClick={() => onEditActionClicked(conversation)}>
+              {conversation.title}
+            </EuiLink>
+          ),
+        },
+        {
+          name: i18n.CONVERSATIONS_TABLE_COLUMN_SYSTEM_PROMPT,
+          align: 'left',
+          render: (conversation: ConversationTableItem) => (
+            <SystemPromptColumn allSystemPrompts={allSystemPrompts} conversation={conversation} />
+          ),
+        },
+        {
+          field: 'actionType',
+          name: i18n.CONVERSATIONS_TABLE_COLUMN_CONNECTOR,
+          align: 'left',
+          render: (actionType: ConversationTableItem['actionType']) =>
+            actionType ? <EuiBadge color="hollow">{actionType}</EuiBadge> : null,
+        },
```

Comment:
We should be able to show the actual Connector Name for Preconfigured connectors here -- I don't think there's much value in knowing it's a Preconfigured connector vs it's actual name. We have a helper for that over here I believe:

<img width="581" alt="image" src="https://github.com/elastic/kibana/assets/2946766/f1834193-0e05-4980-849d-d59b2808a7c8">

https://github.com/elastic/kibana/blob/ce384662aece9ea8c046af93ad6511ce65ad4bba/x-pack/packages/kbn-elastic-assistant/impl/connectorland/helpers.tsx#L63-L76

Reply:
Right! It's retrieved from `connectorTypeTitle ` : https://github.com/elastic/kibana/pull/184678/files#diff-349f3e30d7f72c0a1fcdf1579a65977cd2b0f2636e83552219c51596577d8de9R128
I think I should rename this field to `connectorTypeTitle`